### ISSUE TITLE
Size of inode ... does not agree with number of blocks

### ISSUE THREAD

**Issue by szepeviktor** [s3ql/s3ql#186]
The latest version was introduced on all of my servers on Python 3.7.

```
WARNING: Size of inode 39 (/db-system/mysql-mysql.sql) does not agree with number of blocks, setting from 3577428 to 3579904
WARNING: Size of inode 40 (/db-system/mysql-information_schema.sql) does not agree with number of blocks, setting from 3867707 to 3870720
WARNING: Size of inode 41 (/db-system/mysql-performance_schema.sql) does not agree with number of blocks, setting from 74217 to 77824
WARNING: Size of inode 12451877 (/etc/4/debconf.selections) does not agree with number of blocks, setting from 52517 to 53248
WARNING: Size of inode 12451878 (/etc/4/packages.selections) does not agree with number of blocks, setting from 18512 to 20480
WARNING: Size of inode 12535821 (/etc/5/debconf.selections) does not agree with number of blocks, setting from 52517 to 53248
WARNING: Size of inode 12535822 (/etc/5/packages.selections) does not agree with number of blocks, setting from 18512 to 20480
```

Could you help me understand what does it mean? How does it occur?
https://github.com/s3ql/s3ql/blob/966dab4f1e735991110587dfb16d7d4aff3495dd/src/s3ql/fsck.py#L498-L500
Is it an error?

**Comment by Nikratio** [comment]
This means that there is an internal inconsistency in the metadata. It should only happen when the filesystem was not properly unmounted (otherwise it indicates a bug in S3QL).

The condition that is being checked is that the stored size of a file is equal to at least [number of blocks that are allocated to the file minus 1] x [blocksize] + [size of last block].

**Comment by szepeviktor** [comment]
Thank you.
I reopen this if it keeps coming.

**Comment by szepeviktor** [comment]
@Nikratio Is it the bug fixed in https://github.com/s3ql/s3ql/releases/tag/release-3.7.3 ?

**Comment by Nikratio** [comment]
There's no way to say for certain with the above information. 

However, if the new sizes in the message above are the old sizes rounded up to next multiple of 512, and if you previously got messages of the form `WARNING: Writing dirty block <X> of inode <Y>` for the same inodes, then I'd say the probability is very high that you got hit by the bug that is fixed by the 3.7.3 release.

**Comment by szepeviktor** [comment]
Thank you.

All numbers above are N*512.
https://github.com/s3ql/s3ql/issues/186#issue-646597833
